src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VMDynamicTests.ps1

```diff
@@ -0,0 +1,341 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+function get_all_vm_locations
+{
+    $locations = Get-AzureLocation | where { $_.Name -like 'Microsoft.Compute/virtualMachines' } | select -ExpandProperty Locations;
+
+    for ($i = 0; $i -lt $locations.Count; $i++)
+    {
+        $locations[$i] = ($locations[$i] -Replace ' ', '').ToLower();
```

Comment:
If a cmdlet cannot handle a standard location name, I consider it a bug.  A customer should not be worried about canonical form and must be able to use a location name from Get-AzureLocation.  (Is there any document that a customer should convert a location name to a canonical form for a certain case?)  So, any test should not convert a standard location name into a canonical name.  It should be handled by cmdlet (or by client library or by server.)